### DISCUSSION TITLE
Error in using TwoSampleMR package

### DISCUSSION THREAD

**Discussion by object-notfound** [MRCIEU/TwoSampleMR#346] [Q&A] [answered]
When I was using TwoSampleMR package, after extracing exposure data, it pointed out that "Warning message: In format_data(as.data.frame(exposure_dat), type = "exposure", snps = NULL, "
It directly led to the failure of MR analysing.
The code and problem are in the image as follows.
I will be very appreciate if there is anyone know how to tackle this problem. Thank you very much! :-)
![q1](https://user-images.githubusercontent.com/104241076/164880250-e81d3b07-c474-4a3c-9cc5-e85e14471b6e.png)

**Comment by remlapmot** [comment, marked as answer]
This is a warning message, not an error. The warning tells you that you have not specified the `phenotype_col` argument to `read_exposure_data()` but it has used `'exposure'` in any case.

More importantly your lines 1 and 2 are not correct. In both cases the `t2d_file` object is empty character string because you are using the wrong function to read in your .xlsx or .csv file,

``` r
t2d_file <- system.file("extdata/euro-t2d.xlsx")
print(t2d_file)
#> [1] ""
t2d_file <- system.file("extdata/euro-t2d.csv", package = "TwoSampleMR")
print(t2d_file)
#> [1] ""
```

<sup>Created on 2022-04-25 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

i.e., do not use `system.file()`; to read in the .csv file use say `read.csv()` from the **read.table** package and there are similar functions to read in .xlsx files.

**Reply by object-notfound** [reply]
Thank you very much! Now I understand what the problem is, and I will try the functions you mentioned to  read .xlsx and .csv file.
